### PR TITLE
[move-lang] allow specifying directories for input files and dependencies

### PR DESCRIPTION
**Pull request by bob-wilson**
When compiling a Move file that is not part of the standard library, there are typically a number of dependencies on the stdlib modules. It is a hassle to specify these individually. This change lets you specify the directory containing the stdlib modules as a dependency, and the compiler will expand that to include all the ".move" files inside that directory. We can do the same for the input files to make it easier to build the stdlib (or any other collection of Move files).

This also changes the move-build and move-check programs to customize the error printing. The default output for io::Error results is not very pretty.

## Motivation

Compiler ease of use

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

Ran existing tests.

It would be nice to exercise this new functionality somewhere in our testing process, but I'm not sure where to do that. Suggestions welcome.

### CODE REVIEW

File: language/move-lang/src/lib.rs [hunk 1]
```diff
@@ -29,9 +29,12 @@ use shared::Address;
 use std::{
     collections::HashMap,
     fs::File,
-    io::{self, Read, Write},
+    io::{self, ErrorKind, Read, Write},
+    path::Path,
 };
 
+const MOVE_EXTENSION: &str = "move";
```
Comment (tnowacki) [inline language/move-lang/src/lib.rs:36]: Fine for now, but I do have this extension listed in a test util somewhere. Should try to get it down to a single constant at some point

File: language/move-lang/src/bin/move-build.rs [hunk 2]
```diff
@@ -47,7 +47,17 @@ pub struct Options {
...
+    std::process::exit(match real_main() {
+        Ok(_) => 0,
+        Err(e) => {
+            eprintln!("error: {}", e);
+            1
+        }
+    });
+}
+
+fn real_main() -> std::io::Result<()> {
```
[the remark this answers is not in the document]
Comment (bob-wilson) [reply language/move-lang/src/bin/move-build.rs]: OK

File: language/move-lang/src/lib.rs (hunk 1 above)
Comment (bob-wilson) [reply language/move-lang/src/lib.rs:36]: Agreed

**Pull request closed without merging**